src/coreclr/src/debug/createdump/crashinfomac.cpp

```diff
@@ -349,37 +348,35 @@ CrashInfo::ReadProcessMemory(void* address, void* buffer, size_t size, size_t* r
     // and the size be a multiple of the page size.  We can't differentiate
     // between the cases in which that's required and those in which it
     // isn't, so we do it all the time.
-    int* addressAligned = (int*)((SIZE_T)address & ~(PAGE_SIZE - 1));
-    ssize_t offset = ((SIZE_T)address & (PAGE_SIZE - 1));
+    vm_address_t addressAligned = (vm_address_t)address & ~(PAGE_SIZE - 1);
```

Comment:
I wonder - cannot we just use vm_copy instead of all of this code?

Reply:
Ah, I am sorry, I've misread the doc on the vm_copy, thinking it is a cross process copy.